src/main/groovy/nextflow/trace/WebLogObserver.groovy

```diff
@@ -0,0 +1,229 @@
+ *
+ *   Nextflow is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Nextflow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nextflow.trace
+
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import groovy.util.logging.Slf4j
+import groovyx.gpars.agent.Agent
+
+import nextflow.Session
+import nextflow.script.WorkflowMetadata
+import nextflow.util.SimpleHttpClient
+import nextflow.processor.TaskHandler
+
+/**
+ * Send out messages via HTTP to a configured URL on different workflow
+ * execution events.
+ *
+ * @author Sven Fillinger <sven.fillinger@qbic.uni-tuebingen.de>
+ */
+@Slf4j
+class WebLogObserver implements TraceObserver{
+
+    /**
+     * Workflow identifier, will be taken from the Session() object later
+     */
+    private String runName
+
+    /**
+     * Store the sessions unique ID for downstream reference purposes
+     */
+    private String runId
+
+    /**
+     * Not a HTTP header, but a message header with some general workflow info
+     */
+    private static JsonSlurper JSLURPER = new JsonSlurper()
+
+    /**
+     * The default url is localhost
+     */
+    static String DEF_URL = 'http://localhost'
+
+    /**
+     * Contains server request response
+     */
+    String response = ""
+
+    /**
+     * Simple http client object that will send out messages
+     */
+    private SimpleHttpClient httpClient = new SimpleHttpClient()
+
+    /**
+     * Holds information of workflow metadata
+     */
+    private WorkflowMetadata workflowMetadata
+
+    /**
+     * An agent for the http request in an own thread
+     */
+    private Agent<WebLogObserver> webLogAgent
+
+    /**
+     * Constructor that consumes a URL and creates
+     * a basic HTTP client.
+     * @param url The target address for sending messages to
+     */
+    WebLogObserver(String url) {
+        this.httpClient.setUrl(checkUrl(url))
+        this.runName = ""
+        this.runId = ""
+        this.webLogAgent = new Agent<>(this)
+    }
+
+    /**
+     * Check the URL and create an HttpPost() object. If a invalid i.e. protocol is used,
+     * the constructor will raise an exception.
+     *
+     * The RegEx was taken and adapted from http://urlregex.com
+     *
+     * @param url String with target URL
+     * @return The requested url or the default url, if invalid
+     */
+    static String checkUrl(String url){
+        try {
+            assert url=~ "^(https|http)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]"
+        } catch (AssertionError e){
+            log.error "The given URL $url seems not to be a valid HTTP URL. Only HTTP or HTTPS protocols are supported"
+            url = ""
+        }
```

Comment:
Better to replace the try/catch with a check that matches the url pattern and throw a IllegalArgumentException it not.

Reply:
I wanted to not let the feature not break the workflow run, but I am happy to throw Exceptions :)

Reply:
Good point, but if the url is invalid,I think it's better to about the execution and report it. For temporary connection error, that's a good idea just to log them.